### PR TITLE
only use loadbalaner reference if needed

### LINKED ISSUE #974: ManagedAPIServerLoadBalancer flag doesn't work as expected
/kind bug

**What steps did you take and what happened:**
We don’t have loadbalancer in our Openstack installation. If I set `ManagedAPIServerLoadBalancer: false` to disable load balancer usage I see an error `failed to create load balancer service client: No suitable endpoint could be found in the service catalog.` in logs and reconciliation fails.

**What did you expect to happen:**
I expected to see successful reconciliation without load balancer errors. I think that if I set `ManagedAPIServerLoadBalancer: false`, controller shouldn’t try to create loadBalancerService.

**Anything else you would like to add:**
Discussion in slack: https://kubernetes.slack.com/archives/CFKJB65G9/p1628930844032400

**Environment:**

- Cluster API Provider OpenStack version: v0.4.0
- Cluster-API version: v0.4.0
- OpenStack version: N/A
- Minikube/KIND version: v0.11.1
- Kubernetes version (use `kubectl version`): v1.21.1
- OS (e.g. from `/etc/os-release`): 20.04.2 LTS (Focal Fossa)

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #974

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: controllers/openstackcluster_controller.go [hunk 1]
```diff
@@ -365,16 +365,19 @@ func reconcileBastion(log logr.Logger, osProviderClient *gophercloud.ProviderCli
 }
 
 func reconcileNetworkComponents(log logr.Logger, osProviderClient *gophercloud.ProviderClient, clientOpts *clientconfig.ClientOpts, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster) error {
```
Comment (tobiasgiese) [inline controllers/openstackcluster_controller.go:367]: You could remove the initialization of this variable, as it‘s only used once

File: controllers/openstackcluster_controller.go [hunk 2]
```diff
@@ -365,16 +365,19 @@ func reconcileBastion(log logr.Logger, osProviderClient *gophercloud.ProviderCli
...
+	var loadBalancerService *loadbalancer.Service
```
Comment (tobiasgiese) [inline controllers/openstackcluster_controller.go:368]: You could remove the initialization of this variable, as it‘s only used once

File: controllers/openstackcluster_controller.go [hunk 3]
```diff
@@ -365,16 +365,19 @@ func reconcileBastion(log logr.Logger, osProviderClient *gophercloud.ProviderCli
...
 	}
 
-	loadBalancerService, err := loadbalancer.NewService(osProviderClient, clientOpts, log)
-	if err != nil {
-		return err
+	if openStackCluster.Spec.ManagedAPIServerLoadBalancer {
+		loadBalancerService, err = loadbalancer.NewService(osProviderClient, clientOpts, log)
+		if err != nil {
+			return err
+		}
```
[the remark this answers is not in the document]
Comment (jichenjc) [reply controllers/openstackcluster_controller.go]: updated, thanks~

File: controllers/openstackcluster_controller.go [hunk 4]
```diff
@@ -365,18 +365,14 @@ func reconcileBastion(log logr.Logger, osProviderClient *gophercloud.ProviderCli
 }
 
 func reconcileNetworkComponents(log logr.Logger, osProviderClient *gophercloud.ProviderClient, clientOpts *clientconfig.ClientOpts, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster) error {
+	var loadBalancerService *loadbalancer.Service
```
Comment (tobiasgiese) [inline controllers/openstackcluster_controller.go:368]: You can remove the initialization, as it‘s only used once inside the condition :)

Comment (jichenjc) [reply controllers/openstackcluster_controller.go:368]: right, forgot this ..

**Review by seanschneeweiss — COMMENTED** [review]
/lgtm

**Pull request merged**